posts/2023-07-13-Rust-1.71.0.md

```diff
@@ -0,0 +1,131 @@
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.71.0 stable
+
+### C-unwind ABI
+
+1.71.0 stabilizes `C-unwind` (and other `-unwind` suffixed ABI variants[^1]).
+
+The behavior for unforced unwinding (the typical case) is specified in [this
+table from the RFC which proposed this feature][rfc-table]. To summarize:
+
+Each ABI is mostly equivalent to the same ABI without `-unwind`, except that
+with `-unwind` the behavior is defined to be safe when an unwinding operation
+(`panic` or C++ style exception) crosses the ABI boundary. For `panic=unwind`,
+this is a valid way to let exceptions from one language unwind the stack in
+another language without terminating the process (as long as the exception is
+caught in the same language from which it originated); for `panic=abort`, this
+will typically abort the process immediately.
+
+For this initial stabilization, *no change* is made to the existing ABIs (e.g.
+`"C"`), and unwinding across them remains undefined behavior. A future Rust
+release will amend these ABIs to match the behavior specified in the RFC as the
+final part in stabilizing this feature (usually aborting at the boundary).
+Users are encouraged to start using the new unwind ABI variants in their code
+to remain future proof if they need to unwind across the ABI boundary.
+
+### Debugger visualization attributes
+
+1.71.0 stabilizes support for a new attribute, `#[debug_visualizer(natvis_file
+= "...")]` and `#[debug_visualizer(gdb_script_file = "...")]`, which allows
+embedding Natviz descriptions and GDB scripts into Rust libraries to
+improve debugger output when inspecting data structures created by those
+libraries. Rust itself has packaged similar scripts for some time for the
+standard library, but this feature makes it possible for library authors to
+provide a similar experience to end users.
+
+See the [reference](https://doc.rust-lang.org/nightly/reference/attributes/debugger.html#the-debugger_visualizer-attribute)
+for details on usage.
+
+### raw-dylib linking
+
+On Windows platforms, Rust now supports using functions from dynamic libraries without requiring those libraries to be available at build time, using the new `kind="raw-dylib”` option for `#[link]`.
+
+This avoids requiring users to install those libraries (particularly difficult for cross-compilation), and avoids having to ship stub versions of libraries in crates to link against. This simplifies crates providing bindings to Windows libraries.
+
+Rust also supports binding to symbols provided by DLLs by ordinal rather than named symbol, using the new `#[link_ordinal]` attribute.
```

Comment:
Sorry I'm late to this, but I think people are finding this confusing due to the unclear distinction between types of libraries. You've never needed to install the `.dll` (dynamic library) in order to link to it. In fact having it is no help at all. You did need a `.lib` (a static library) that describes which symbol is imported from which dll. This is usually called an "import library". But now you don't need it because `raw-dylib`.

Admittedly I don't have better wording right now but I thought I should bring it up.

Reply:
@ChrisDenton With hindsight, it might have been clearer to say "This avoids requiring *developers* to install those libraries", rather than "*users*".